### DISCUSSION TITLE
Why I can't process the .env.local?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#37488] [Help] [answered]
I am getting this error here:

```
 if (process.env.NEXT_PUBLIC_MONGODB_URI === "development") {
error - utils\mongodb.js (10:8) @ eval
Error: Please add your Mongo URI to .env.local
   8 | 
   9 | if (!process.env.NEXT_PUBLIC_MONGODB_URI) {
> 10 |   throw new Error("Please add your Mongo URI to .env.local");
     |        ^
  11 | }
  12 | 
  13 | if (process.env.NEXT_PUBLIC_MONGODB_URI === "development") {
```

The error reads what it reads because that is what is written to read... but, why isn't the .env being processed?

The URI is good.  I connect just fine to my mongo shell, and other apps are working fine with the same string.

I am about to give up trying to pull external data....  could I just upload the data in a local json and work from there?

The page that I am trying to put the data in is this one:
https://github.com/Elindo586/nextprojectb/blob/main/pages/inventory.js

The connection is this one:
https://github.com/Elindo586/nextprojectb/blob/main/utils/mongodb.js

**Comment by ephraimduncan** [comment]
When you edit the `.env.local` file, you have to restart the server for the Next.js to read the environmental variable.

**Comment by icyJoseph** [comment]
Hi,

If restarting the application was not the problem, which it could totally be. Then, it might be a typo or something odd like that, or perhaps there's an `.env.development.local` file? which sets `NEXT_PUBLIC_MONGODB_URI` to `""`, and that file would take precedence and not override variables declare by `.env.local`.

```
Depending on the environment (as set by NODE_ENV), Environment Variables 
are loaded from the following sources in top-to-bottom order. In all environments,
the existing env is not overridden by following sources:

NODE_ENV=production
.env.production.local
.env.local
.env.production
.env

NODE_ENV=development
.env.development.local
.env.local
.env.development
.env

NODE_ENV=test
.env.test.local
.env.test
.env
```

**Comment by Elindo586** [comment]
So, restarting the server didn't do anything.

I only have one .env.local  which located at root/utils/.env.local

The actual connection file that calls the .env.local is located at  root/utils/mongodb.js

And then the page calling for the connection is under pages.  inventory.js

The error comes when I visit the inventory page that triggers the connection.

The console gives me a bunch of webpack errors....

```
index.js?46cb:365 Uncaught     at eval (webpack-internal:///./utils/mongodb.js:13:11)
    at Object../utils/mongodb.js (file://C:\Users\Edgar\Desktop\Stuff\Acode\nextprojectb\.next\server\pages\inventory.js:143:1)
    at __webpack_require__ (file://C:\Users\Edgar\Desktop\Stuff\Acode\nextprojectb\.next\server\webpack-runtime.js:33:42)
    at eval (webpack-internal:///./pages/inventory.js:16:72)
    at Object../pages/inventory.js (file://C:\Users\Edgar\Desktop\Stuff\Acode\nextprojectb\.next\server\pages\inventory.js:132:1)
    at __webpack_require__ (file://C:\Users\Edgar\Desktop\Stuff\Acode\nextprojectb\.next\server\webpack-runtime.js:33:42)
    at __webpack_exec__ (file://C:\Users\Edgar\Desktop\Stuff\Acode\nextprojectb\.next\server\pages\inventory.js:416:39)
    at <unknown> (file://C:\Users\Edgar\Desktop\Stuff\Acode\nextprojectb\.next\server\pages\inventory.js:417:28)
    at Object.<anonymous> (file://C:\Users\Edgar\Desktop\Stuff\Acode\nextprojectb\.next\server\pages\inventory.js:420:3)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
```

**Reply by icyJoseph** [reply]
but the .env vars ought to be at the top level, as a sibling to the other folders, sort of like this:

```
├── .env.local
├── README.md
├── jest.config.js
├── next-env.d.ts
├── next.config.js
├── package.json
├── pages
├── public
└── yarn.lock
```

**Comment by Elindo586** [comment]
BTW... the URI in the .env looks like this:

NEXT_PUBLIC_MONGODB_URI=mongodb+srv://<username>:<password>@cluster0.gcyyo.mongodb.net/?retryWrites=true&w=majority

including ...

...srv://myusername:mypassword@cluster0....

**Comment by Elindo586** [comment, marked as answer]
├── .env.local
├── README.md
├── jest.config.js
├── next-env.d.ts
├── next.config.js
├── package.json
├── pages
├── public
└── yarn.lock


^^^  this made the error go away...

**Comment by mfrezghi** [comment]
The thing that fixed my error was changing all my file types to the same .js file. My route file was a TypeScript file, and my database file was a JavaScript file. Switching my route file to js resolved my issue.

**Comment by Marcelovv2** [comment]
I fixed my by going from .env.local.example to .env.local
